File: internal/publish/publicacao.go
package publish

import "time"

// Publicacao representa uma publicação agendada ou já executada.
// Segue o mesmo padrão append-only do modelo de Busca.
type Publicacao struct {
	ID         string    `json:"id"`
	ProdutoID  string    `json:"produto_id"`
	Nome       string    `json:"nome"`
	Categoria  string    `json:"categoria"`
	Preco      float64   `json:"preco"`
	Comissao   float64   `json:"comissao"`
	Link       string    `json:"link"`
	Imagem     string    `json:"imagem"`
	Estrategia string    `json:"estrategia"`
	DestinoID  string    `json:"destino_id"`
	TemplateID string    `json:"template_id"`
	AgendadaEm string    `json:"agendada_em"` // ISO 8601 — quando deve ser enviada (vazio = imediata)
	Status     string    `json:"status"`      // "agendada" | "enviada" | "erro"
	Detalhe    string    `json:"detalhe"`     // mensagem de erro ou sub_id
	CriadaEm   time.Time `json:"criada_em"`
	EnviadaEm  string    `json:"enviada_em,omitempty"` // ISO 8601
	OwnerUID   string    `json:"owner_uid,omitempty"`
}

File: src/Garimpei.Domain/Entities/Publicacao.cs
using Garimpei.Domain.Interfaces;

namespace Garimpei.Domain.Entities;

/// <summary>
/// Publicação agendada ou enviada em um canal.
/// </summary>
public sealed class Publicacao : IOwnedEntity
{
    public Guid Id { get; init; } = Guid.NewGuid();
    public required string ProdutoId { get; init; }
    public required string Nome { get; init; }
    public string? Categoria { get; set; }
    public decimal Preco { get; set; }
    public double Comissao { get; set; }
    public string? Link { get; set; }
    public string? Imagem { get; set; }
    public string? Estrategia { get; set; }
    public string? DestinoId { get; set; }
    public string? TemplateId { get; set; }
    public DateTime? AgendadaEm { get; set; }
    public string Status { get; set; } = "pendente"; // pendente | enviada | erro
    public string? Detalhe { get; set; }
    public string OwnerUid { get; set; } = string.Empty;
    public DateTime CreatedAt { get; init; } = DateTime.UtcNow;
    public DateTime? EnviadaEm { get; set; }
}
